### DISCUSSION TITLE
[tasks] clock drift detected ... woke up ... sleeping until ... to correct clock

### DISCUSSION THREAD

**Discussion by aurkaxi** [Rapptz/discord.py#9038] [General]
![unknown](https://user-images.githubusercontent.com/72933395/199307466-00df5fd0-7069-42fd-82f5-e773ee63efe6.png)
getting tons of this warning just before the task actually happens. my task would be run in 12:00 am. but it woke up at 11:59 pm and keep saying "sleeping until to correct clock"
my code:
```py
    @tasks.loop(time=time(0, 0, 0, tzinfo=myzone))
    async def update(self):
        #verify first day of monthh
        now = datetime.now(myzone)
        if now.day == 1:
            print("Updating monthly leaderboard")
        else:
            print("Not updating monthly leaderboard")
```

**Comment by CaedenPH** [comment]
Not sure if this is the recommended answer, but I'd personally use a asycnhronous `crontab` library such as
https://pypi.org/project/aiocron/
For example,
```py
@aiocron.crontab('0 0 * * *')
async def attime():
    print('run')
```
Would run every day at `00:00`

**Reply by aurkaxi** [reply]
i'd just stick with dpy, because it has it built in

**Reply by Rapptz** [reply]
Not sure why you think this library would be impervious to clock drifting.

**Comment by Rapptz** [comment]
That warning is because the task woke up earlier than the expected time and needs to correct itself. It usually means either asyncio woke up too early or your clock is out of sync. There's nothing really I can do here other than print a warning to tell you that something is up with your clock and try to correct it. It is a tad bit odd to see this print more than once though.

**Reply by aurkaxi** [reply]
why did task woke up? i don't understand why my clock would act werid. it's just datetime lib. is there anything i can do from my end?

**Comment by kaumaron** [comment]
I'm getting a similar warning as well. I have two time scheduled tasks that pop this twice warning twice everyday since moving into Docker. It didn't do it beforehand but started in Docker. My container is running in UTC and the task is scheduled in Americas/New_York. That might be related.
